### PR TITLE
fix(rest): Download licenseinfo file rest endpoint.

### PR DESCRIPTION
**Pull request by smrutis1**
* Fixed the issue with download license information file rest endpoint and added the documentation for the same. 

closes: #575 

Signed-off-by: Smruti Sahoo <smruti.sahoo@siemens.com>

### CODE REVIEW

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 1]
```diff
@@ -283,12 +283,13 @@ public void downloadLicenseInfo(@PathVariable("id") String id,
         final String projectName = sw360Project.getName();
         final String projectVersion = sw360Project.getVersion();
 	final String timestamp = SW360Utils.getCreatedOnTime().replaceAll("\\s", "_").replace(":", "_");
-	final OutputFormatInfo outputFormatInfo = licenseInfoService.getOutputFormatInfoForGeneratorClass(generatorClassName);
+	String[] outputGeneratorClassNameWithVariant = generatorClassNameWithVariant.split("::");
+	final OutputFormatInfo outputFormatInfo = licenseInfoService.getOutputFormatInfoForGeneratorClass(outputGeneratorClassNameWithVariant[0]);
```
Comment (nutanv1contr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java]: Please remove white spaces.

File: rest/resource-server/src/docs/asciidoc/projects.adoc [hunk 2]
```diff
@@ -247,3 +247,16 @@ include::{snippets}/should_document_link_releases/curl-request.adoc[]
...
 
+[[resources-project-get-download-licenseinfo]]
+==== Download License Info
+
+A `GET` request is used to download license info for the project.
+
+===== Example request
+include::{snippets}/should_document_get_download_license_info/curl-request.adoc[]
+
+===== Example request parameter
```
Comment (nutanv1contr) [inline rest/resource-server/src/docs/asciidoc/projects.adoc]: Could you document "Request parameters" before "Example request" as it is done for other rest end point documentation.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java (hunk 1 above)
Comment (nutanv1contr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java]: Could you rename the field generatorClassName to generatorClassNameWithVariantinside inside licenseInfoService.getLicenseInfoFile(..) method also.

**Comment by nutanv1contr** [comment]
Tested the rest end point. Now License info is returned as response to the rest request. Generate license info project feature is now working from rest end point.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java (hunk 1 above)
Comment (smrutis1) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java]: > Please remove white spaces.

Removed it

Comment (smrutis1) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java]: > Could you rename the field generatorClassName to generatorClassNameWithVariant inside licenseInfoService.getLicenseInfoFile(..) method also.

Done

File: rest/resource-server/src/docs/asciidoc/projects.adoc (hunk 2 above)
Comment (smrutis1) [reply rest/resource-server/src/docs/asciidoc/projects.adoc]: Updated the order.

**Comment by nutanv1contr** [comment]
Code looks good to me now.

**Comment by maxhbr** [comment]
> Code looks good to me now.

I have removed the `needs code review` label


> Tested the rest end point. Now License info is returned as response to the rest request. Generate license info project feature is now working from rest end point.

I have removed the `needs general test` label

**Comment by mcjaeger** [comment]
(from discussion on slack) small change discussed to make two parameters instead of a combined one.

**Comment by smrutis1** [comment]
Updated the PR by splitting the combined request parameter from `generatorClassNameWithVariant` to `generatorClassName` and `variant` and updated the rest documentation accordingly.

**Comment by mcjaeger** [comment]
test restarted, although, maybe there is a real (not a timing) issue:
```
Running org.eclipse.sw360.cvesearch.entitytranslation.CveSearchDataTranslatorTest
Tests run: 5, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 350.654 sec <<< FAILURE!
testWithApacheData(org.eclipse.sw360.cvesearch.entitytranslation.CveSearchDataTranslatorTest)  Time elapsed: 326.111 sec  <<< ERROR!
com.google.gson.JsonSyntaxException: java.io.EOFException: End of input at line 1782 column 5 path $[16].vulnerable_configuration_cpe_2_2[78]
	at com.google.gson.Gson.fromJson(Gson.java:897)
	at com.google.gson.Gson.fromJson(Gson.java:852)
```

**Comment by mcjaeger** [comment]
seems like unit testing fails for cve search fails at three branches now ...

**Comment by mcjaeger** [comment]
I think we can remove code review but need a small test? (is this correctly understood?)

**Comment by nutanv1contr** [comment]
Code looks good.We can remove the code review.

**Comment by nutanv1contr** [comment]
Tested the PR again.Functionality is working as expected.

**Pull request merged**